### PR TITLE
Local kolla container image builds

### PR DESCRIPTION
**Pull request by m-bull**
- Sync all repositories to local Pulp

### CODE REVIEW

**Comment by priteau** [comment]
Don't forget to add the following to `etc/kayobe/containers/pulp/settings.py`:

```
ALLOWED_CONTENT_CHECKSUMS = ["sha1", "sha224", "sha256", "sha384", "sha512"]
```

File: etc/kayobe/pulp.yml [hunk 1]
```diff
@@ -317,6 +322,106 @@ stackhpc_pulp_repository_rpm_repos:
...
 
+  # Additional repositories required to build Kolla containers from local Pulp service.
+  - name: CentOS Stream 8 - OpenStack Wallaby
+    url: "{{ stackhpc_release_pulp_content_url }}/centos/8-stream/cloud/x86_64/openstack-wallaby/{{ stackhpc_pulp_repo_centos_stream_8_openstack_wallaby_version }}"
+    client_cert: "{{ stackhpc_release_pulp_client_cert }}"
+    client_key: "{{ stackhpc_release_pulp_client_key }}"
+    policy: on_demand
+    sync_policy: mirror_complete
+    state: present
+    required: "{{ stackhpc_pulp_sync_for_local_container_build and stackhpc_pulp_sync_el_8 | bool }}"
```
Comment (markgoddard) [inline etc/kayobe/pulp.yml]:
```suggestion
    required: "{{ stackhpc_pulp_sync_for_local_container_build | bool and stackhpc_pulp_sync_el_8 | bool }}"
```

**Review by markgoddard — COMMENTED** [review]
Thanks! Reno pls. I was going to ask for docs but we don't really have anything to add to in this area.

File: releasenotes/notes/kolla-local-builds-eca0277a969803fd.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+features:
+  - |
+    Adds a new variable ``stackhpc_pulp_sync_for_local_container_build`` which,
+    when set to ``true``, configures the local Pulp server to sync all package
+    repositories required for building kolla containers on a local kolla build
+    host.
```
Comment (priteau) [inline releasenotes/notes/kolla-local-builds-eca0277a969803fd.yaml]: Nit: add new line.

**Review by priteau — CHANGES_REQUESTED** [review]

File: etc/kayobe/containers/pulp/settings.py [hunk 3]
```diff
@@ -2,3 +2,6 @@
 ANSIBLE_API_HOSTNAME='{{ pulp_url }}'
 ANSIBLE_CONTENT_HOSTNAME='{{ pulp_url }}/pulp/content'
 TOKEN_AUTH_DISABLED=True
+{% if stackhpc_pulp_sync_for_local_container_build | bool %}
+ALLOWED_CONTENT_CHECKSUMS = ["sha1", "sha224", "sha256", "sha384", "sha512"]
+{% endif %}
```
Comment (priteau) [inline etc/kayobe/containers/pulp/settings.py]: Nit: add new line

File: etc/kayobe/pulp.yml [hunk 4]
```diff
@@ -317,6 +322,106 @@ stackhpc_pulp_repository_rpm_repos:
     state: present
     required: "{{ stackhpc_pulp_sync_el_8 }}"
 
+  # Additional repositories required to build Kolla containers from local Pulp service.
+  - name: CentOS Stream 8 - OpenStack Wallaby
+    url: "{{ stackhpc_release_pulp_content_url }}/centos/8-stream/cloud/x86_64/openstack-wallaby/{{ stackhpc_pulp_repo_centos_stream_8_openstack_wallaby_version }}"
+    client_cert: "{{ stackhpc_release_pulp_client_cert }}"
```
Comment (priteau) [inline etc/kayobe/pulp.yml]: You need to switch to username and password instead of certificates.

**Review by priteau — CHANGES_REQUESTED** [review]

File: etc/kayobe/pulp.yml (hunk 4 above)
Comment (priteau) [reply etc/kayobe/pulp.yml]: I pushed this as an additional commit.

**Review by markgoddard — COMMENTED** [review]
LGTM

**Comment by priteau** [comment]
@m-bull Do you consider this ready for review now?

File: etc/kayobe/pulp.yml (hunk 4 above)
Comment (m-bull) [reply etc/kayobe/pulp.yml]: Thanks @priteau

**Review by markgoddard — APPROVED** [review]

**Review by priteau — APPROVED** [review]
LGTM although I've not tested this latest version.

**Pull request merged**